library/Vanilla/Models/FullRecordCacheModel.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license Proprietary
+ */
+
+namespace Vanilla\Models;
+
+/**
+ * A model that caches all queries that come through, and invalidates the cache on every change.
+ *
+ * Ideal for DB tables that are infrequently updated and have smaller record sets.
+ */
+class FullRecordCacheModel extends PipelineModel {
+
+    /** @var ModelCache $cache */
+    protected $modelCache;
+
+    /**
+     * Model constructor.
+     *
+     * @param string $table Database table associated with this resource.
+     * @param \Gdn_Cache $cache The cache instance.
+     * @param array|null $defaultCacheOptions Default options to apply for storing cache values.
+     */
+    public function __construct(string $table, \Gdn_Cache $cache, ?array $defaultCacheOptions = []) {
```

Comment:
Do you think this might be easier with `array` instead of `?array`?

Reply:
Done.